storage/filesystem/fat/source/FATFileSystem.cpp

```diff
@@ -404,7 +404,15 @@ int FATFileSystem::format(BlockDevice *bd, bd_size_t cluster_size)
     }
 
     // Logical drive number, Partitioning rule, Allocation unit size (bytes per cluster)
-    FRESULT res = f_mkfs(fs._fsid, FM_ANY | FM_SFD, cluster_size, NULL, 0);
+    static const MKFS_PARM opt = {
+        FM_ANY | FM_SFD, /* Format option (FM_FAT, FM_FAT32, FM_EXFAT and FM_SFD) */
+        1,               /* Number of FATs */
+        0,               /* Data area alignment (sector) */
+        512,             /* Number of root directory entries */
```

Comment:
Why not leave the default value of 0 for `n_fat` and `n_root` as was done previously?

Reply:
Thanks for looking into this! It's good to have a second set of eyes for this complex matter :)

You mean those?

https://github.com/ARMmbed/mbed-os/blob/860267c8c84a29be1c1713511aae3a555ee107d0/storage/filesystem/fat/ChaN/ff.cpp#L5546-L5547

They are not defined any more inside `FRESULT f_mkfs`:

https://github.com/ladislas/mbed-os/blob/315a22a298268cf65a857651c6e630589c2678c5/storage/filesystem/fat/ChaN/ff.cpp#L5871

Reply:
@evedon any feedback on my last message?

Reply:
I meant that according to http://elm-chan.org/fsw/ff/doc/mkfs.html a default value of 0 for `n_fat` is converted by the library to `1` and similarly a default value of 0 for `n_root` gives 512. So why not leave 0 and let the library set the defaults as was previously done?

Reply:
ah you're right, I missed that. I can change it back to 0. On the other hand, I think it's not bad to explicitly say what we want instead of relying on default behavior that may not be that obvious next time we look into it.

I thought the failures might come from that but now I have no ideas :( I'll keep looking into it when I get time.

Reply:
If you need these explicit default values, then that's fine.